Code/Engine/Texture/Image/ImageFormat.h

```diff
@@ -1,312 +1,322 @@
+
+  /// \brief Block depth of a compressed format. Defaults to 1 for uncompressed formats.
+  static ezUInt32 GetBlockDepth(Enum format, ezUInt32 planeIndex = 0);
+
+  /// \brief Returns the data type represented by a format.
+  static ezImageFormatDataType::Enum GetDataType(Enum format);
+
+  /// \brief Returns true if the format is compressed.
+  static bool IsCompressed(Enum format);
+
+  /// \brief Returns true if the format is a depth format.
+  static bool IsDepth(Enum format);
+
+  /// \brief Returns whether the given format is an sRGB format.
+  static bool IsSrgb(Enum format);
+
+  /// \brief Returns true if the format is a stencil format.
+  static bool IsStencil(Enum format);
+
+  /// \brief Returns the corresponding sRGB format if one exists; otherwise returns the unmodified format.
+  static Enum AsSrgb(Enum format);
+
+  /// \brief Returns the corresponding linear format if one exists; otherwise returns the unmodified format.
+  static Enum AsLinear(Enum format);
+
+  /// \brief Computes the number of blocks in X direction (compressed) or pixels (if uncompressed) for a given width (in pixels).
+  static ezUInt32 GetNumBlocksX(Enum format, ezUInt32 width, ezUInt32 planeIndex = 0);
+
+  /// \brief Computes the number of blocks in Y direction (compressed) or pixels (if uncompressed) for a given height (in pixels).
+  static ezUInt32 GetNumBlocksY(Enum format, ezUInt32 height, ezUInt32 planeIndex = 0);
+
+  /// \brief Computes the number of blocks in Z direction (compressed) or pixels (if uncompressed) for a given height (in pixels).
+  static ezUInt32 GetNumBlocksZ(Enum format, ezUInt32 depth, ezUInt32 planeIndex = 0);
+
+  /// \brief Computes the size in bytes of a row of blocks (compressed) or pixels (if uncompressed) of the given width.
+  static ezUInt64 GetRowPitch(Enum format, ezUInt32 width, ezUInt32 planeIndex = 0);
+
+  /// \brief Computes the size in bytes of a 2D slice of blocks (compressed) or pixels (if uncompressed) of the given width and height.
+  static ezUInt64 GetDepthPitch(Enum format, ezUInt32 width, ezUInt32 height, ezUInt32 planeIndex = 0);
+
+  /// \brief Returns the type of the image format.
+  static ezImageFormatType::Enum GetType(Enum format);
+
+  /// \brief Finds a format matching the given component masks.
+  static ezImageFormat::Enum FromPixelMask(
     ezUInt32 uiRedMask, ezUInt32 uiGreenMask, ezUInt32 uiBlueMask, ezUInt32 uiAlphaMask, ezUInt32 uiBitsPerPixel);
 
-  /// \brief Returns true if the data formats are compatible, i.e. can be copied into one another
-  static bool IsCompatible(Enum left, Enum right);
-
-  /// \brief Returns true if the most high-res miplevel requires block alignment
-  static bool RequiresFirstLevelBlockAlignment(Enum format);
-};
-
-EZ_DEFINE_AS_POD_TYPE(ezImageFormat::Enum);
+  /// \brief Returns the format of a subplane of a given format.
+  static ezImageFormat::Enum GetSubFormat(ezImageFormat::Enum format, ezUInt32 uiPlaneIndex);

```

Comment:
Maybe call it GetPlaneSubFormat, because the current name is more generic. Also somewhere it should be described what a planar format even is, because we now have support for this, but I don't think that many people even know about it.

Reply:
Changed the name and added some more explanation from Christian to the top of the file.